src/components/container/SwellingCalendar/SwellingCalendar.tsx

```diff
@@ -0,0 +1,177 @@
+            if (exists){
+                if (parseISO(swellingEntry.dateRecorded) > parseISO(exists.dateRecorded)){
+                    exists.severity = swellingEntry.severity;
+                    calendarData.set(key, exists);
+                }
+            }
+            else {
+                calendarData.set(key, swellingEntry);
+            }
+        });
+
+        setData(calendarData);
+    }
+
+    function createResultsMap(results : SurveyAnswer[]){
+        var groupedByResult = new  Map<Guid, SwellingDay>();
+
+        results.forEach( (r) => {
+            var exists = groupedByResult.get(r.surveyResultID);
+            var swellingDay = exists ?? {dateRecorded : "", severity : ""};
+
+            if (r.resultIdentifier === props.dateRecordedResultIdentifier && r.answers)
+            {
+                swellingDay.dateRecorded = r.answers[0];
+            }
+
+            if (r.resultIdentifier === props.severityResultIdentifier && r.answers)
+            {
+                var severity = "";
+                switch (r.answers[0]){
+                    case "mild":
+                    case "severity1":
+                        severity = "mild";
+                        break;
+                    case "moderate":
+                    case "severity2":
+                        severity = "moderate";
+                        break;
+                    case "severe":
+                    case "severity3":
+                        severity = "severe";
+                        break;
+                    }
+
+                    swellingDay.severity = severity;
+            }
+
+            groupedByResult.set(r.surveyResultID, swellingDay);
+        });
+
+        return groupedByResult;
+    }
+
+    const renderDay = (year: number, month: number, day?: number): React.JSX.Element => {
+        return <CalendarDay
+            year={year}
+            month={month}
+            day={day}
+            stateConfiguration={stateConfiguration}
+            computeStateForDay={computeStateForDay}
+        />;
+    };
+
+    function pageMonth(newStart: Date) {
+        setMonth(startOfMonth(newStart));
+    } 
+
+
+    useEffect(() => {
+        initialize();
+        MyDataHelps.on("applicationDidBecomeVisible", initialize);
+
+        return () => {
+            MyDataHelps.off("applicationDidBecomeVisible", initialize);
+        }
+    }, []);
+
+
+    if (!data){
+        return <LoadingIndicator/>
+    }
+    else {
+        return <>
+            <DateRangeNavigator intervalType={"Month"} intervalStart={datePagerMonth} onIntervalChange={pageMonth}></DateRangeNavigator>
+            <Calendar className="mdhui-simple-calendar" year={datePagerMonth.getFullYear()} month={datePagerMonth.getMonth()} dayRenderer={renderDay}/>
+            <Button defaultMargin onClick={() => MyDataHelps.startSurvey(props.surveyName)}>{language("log-swelling")} <FontAwesomeSvgIcon icon={faPlus} /></Button>
```

Comment:
I don't think this button should be in here either.  You should be able to add a button below the calendar component when building views or other components that contain this calendar component.